### ISSUE TITLE
Define safe-prior semantics for unmatched group-only terms

### ISSUE THREAD

**Issue by AlexanderFengler** [lnccbrown/HSSM#1225]
## Problem

HSSM's generated `prior_settings="safe"` priors are still unsafe for a genuinely
group-only coefficient, for example:

```python
v ~ 1 + (0 + x | participant_id)
```

There is no common `x`, so the group distribution owns the population coefficient:

```text
x|participant_id ~ Normal(
    mu=Normal(mu=0, sigma=0.25),
    sigma=Weibull(alpha=1.5, beta=0.3),
)
```

With current Bambi:

- effective `noncentered=True` builds `offset * sigma`, drops `mu`, and leaves a free
  `mu` hyperprior disconnected;
- effective `noncentered=False` retains `mu` and builds the intended model;
- hierarchical non-Normal outer priors cannot be non-centered and fail during model
  construction.

This affects unmatched HDDM group intercepts as well as Normal slopes. On current
HSSM `main`, `v` produces the Normal/orphan case; `a`, `z`, `t`, `sv`, `sz`, `st`, and
`p_outlier` can receive Gamma/Beta-family hierarchies that Bambi cannot non-center.

This is distinct from #1224. A matching common term makes the group coefficient a
zero-mean deviation. Here no common counterpart exists, so blindly replacing the
group location with zero changes the scientific model.

## First-principles policy

Bambi first constructs a complete linear predictor and then applies the inverse link:

```text
eta = X beta + Z b
parameter = inverse_link(eta)
```

Population-location ownership is therefore defined on `eta`:

1. **Matching common term exists.** The common coefficient owns the location; every
   generated group term is a zero-mean deviation. This is #1224 behavior.
2. **Exactly one unmatched group term exists for an expression.** That group
   distribution owns the location. Preserve its prior family/hyperparameters and force
   that generated prior to `noncentered=False` until Bambi can retain `mu` correctly.
3. **Multiple unmatched group terms share one exact expression.** No unique owner
   exists. For example, `(0+x|subject) + (0+x|item)` identifies only the sum of the two
   population locations. If HSSM would generate any of those priors, fail actionably
   and require a common `x` or fully explicit ownership constraints.
4. **Explicit priors.** Never semantically rewrite them. If current Bambi cannot build
   the requested effective parameterization faithfully, fail before Bambi rather than
   continuing with a different model.

Matching remains exact Formulae structure through `term.expr.name`. Do not split group
display names or infer symbolic equivalence/design collinearity.

## Link functions

The same ownership rule applies under every link because the ambiguity occurs before
the inverse link.

- **Identity:** coefficients are on the parameter scale. A group-only location of zero
  may be invalid or a support boundary. Canonical omitted/string/Bambi/HSSM identity
  spellings must select the same response-scale/HDDM prior family.
- **Log:** locations are on the log-predictor scale. `exp(mu)` is a reference/median-like
  baseline, not generally the response mean; a zero matched deviation is the neutral
  multiplier 1.
- **Generalized logit:** locations are on generalized-log-odds. `inverse_link(mu)` is a
  bounded reference, not generally the response expectation.
- **Other/custom transformed links:** treat the group location as a predictor-scale
  coefficient and do not apply response bounds to it.

The current unmatched HDDM factory still tests raw link presence, so omitted identity
can select an HDDM Beta/Gamma family while explicit identity selects generic Normal.
This issue will replace that with semantic identity classification.

The existing "bounds are not applied" warning should be emitted only for effective
identity. Under a transformed support-respecting link, leaving `eta` unbounded is
correct; the inverse link owns support.

## Required behavior

- Generated matched terms remain zero-mean and honor requested centering/NC.
- A unique generated unmatched term retains its family/hyperparameters, receives a
  per-prior `noncentered=False`, and warns only when this overrides an effective NC
  request.
- Multiple unmatched terms sharing an expression fail before safe defaults are
  generated unless every competing group prior is explicit.
- Explicit priors are not semantically rewritten beyond existing dict-to-`Prior`
  normalization.
- Under either parameterization, a group-specific `bmb.Prior` without a top-level
  hyperprior fails early with HSSM's actionable error; numeric fixed group
  coefficients remain supported.
- Under effective NC, HSSM accepts only a plain, untruncated built-in Normal with
  hierarchical `sigma`, absent/fixed-all-zero `mu`, and no extra ignored arguments.
  It raises one aggregated error when an explicit prior would be dropped or cannot
  build:
  - Normal + hierarchical `sigma` + free/nonzero `mu`;
  - Normal with stochastic arguments but no hierarchical `sigma`;
  - hierarchical non-Normal outer prior;
  - custom/truncated Normal or additional arguments Bambi's shortcut would ignore.
- A zero-mean Normal with hierarchical `sigma` remains valid under NC.
- A per-prior/component `noncentered=False` remains authoritative.
- Fully explicit centered models with two or more free additive locations for one
  unmatched expression are preserved but receive a likelihood-ridge warning.
- `prior_settings=None` generates nothing but does not bypass explicit-prior fidelity
  checks.

## Commit plan

1. `refactor: share effective group parameterization (#1225)`
2. `fix: reject ambiguous generated group locations (#1225)`
3. `fix: center generated single-owner group priors (#1225)`
4. `fix: choose group-only priors on the effective link scale (#1225)`
5. `fix: fail fast on unrepresentable explicit group priors (#1225)`
6. `fix: diagnose multiple explicit group locations (#1225)`
7. `docs: explain group-only priors across links (#1225)`

## Acceptance matrix

- All generated unmatched DDM group intercepts (`v`, `a`, `z`, `t`, `sv`, `sz`, `st`,
  `p_outlier`) build under model-level centered and NC requests.
- Requested NC uses a visible centered fallback, direct group RVs, connected
  hyperparameters, and no orphan means.
- Identity, log, generalized-logit, representative built-in, custom, and preset links
  are covered.
- Unmatched slopes, intercepts, categorical terms, interactions, transforms, multiple
  grouping factors, exact priors, and wildcards are covered.
- Model/component/per-prior precedence and scalar/vector means are covered.
- Matched #1224 graphs are unchanged.
- Structure-only tests suffice; no sampling is required.

## Related and deferred work

- #1224 — exact common/group matching.
- #1232 — semantic identity links for common-intercept safe priors.
- bambinos/bambi#1003 — upstream `mu + sigma * offset` support.
- #685 — effective-parameterization visibility after this fix.
- #720 — separate no-intercept common categorical cell-mean policy.

- #1268 will track adoption of an eventual Bambi fix for eligible generated Normal
  terms.
- #1269 will track bound-aware identity-linked group-only intercept priors.

Non-Normal non-centering is not part of this issue.

**Comment by AlexanderFengler** [comment]
The #1224 audit found that this follow-up needs to cover unmatched group **intercepts** as well as the numeric-slope example above.

With `prior_settings="safe"` and the default `noncentered=True`, construction probes on current `main` show two classes of failure:

- unmatched Normal group terms (including `v`) build, but their generated free `mu` becomes a disconnected `*_mu` RV;
- unmatched HDDM group intercepts for `a`, `z`, `t`, `sv`, `sz`, `st`, and `p_outlier` receive outer Gamma/Beta-family hierarchical priors and fail at model build with Bambi's `NotImplementedError`, because its non-centered group implementation supports only an outer Normal prior.

So the policy decision here is not only about preserving location semantics: some generated safe defaults are incompatible with HSSM's default parameterization and cannot build at all.

Suggested additions to the acceptance matrix:

- build unmatched group intercepts for every DDM parameter under both parameterizations;
- assert that generated defaults neither create orphan RVs nor select an outer family Bambi cannot non-center;
- cover `p_outlier` and the variability parameters, not only `v/a/z/t`;
- retain explicit-prior authority and produce a term-specific warning/error when the requested parameterization cannot represent it.

This remains intentionally outside #1224 / PR #1227, which fixes only exact common/group matches.

**Comment by AlexanderFengler** [comment]
## Revised recommendation after the full current-main prior audit

The current-main matrix changes one important part of the policy recommendation in this issue: zero-centering every unmatched generated group term is not safe for a location-bearing group-only intercept on a bounded identity-scale DDM parameter.

### Confirmed failure matrix

Structurally unmatched generated terms currently receive location-bearing priors:

- generic/Normal terms: `Normal(mu=Normal(...), sigma=Weibull(...))`;
- HDDM group-only intercepts: parameter-specific outer distributions (`Normal` for `v`; `Gamma` for `a`, `t`, `sv`, `sz`, `st`; `Beta` for `z` and `p_outlier`).

Under Bambi's effective non-centering:

- outer Normal + hierarchical `sigma` builds as `offset * sigma` and drops `mu`, leaving the generated free mean orphaned;
- hierarchical non-Normal outer distributions are unsupported and raise `NotImplementedError`.

This was reproduced for `v`, `a`, `z`, `t`, `sv`, `sz`, `st`, and `p_outlier` on HSSM `main` at `5e8610b9`.

Under centering, all eight build and preserve the location-bearing prior. A per-prior `noncentered=False` override also builds all eight when the model-level setting is `True`.

Blindly replacing those group-only intercepts with mean-zero deviations is not a compatible fix: for identity-linked bounded parameters, zero can be on the support boundary. An `a` probe at zero exhibited pathological analytical likelihood behavior.

### Recommended policy

For **generated structurally unmatched** group terms:

1. Preserve the existing location-bearing prior family and hyperparameters.
2. Force that generated prior to `noncentered=False`.
3. Warn when this overrides an effectively non-centered model, and identify the exact missing common Formulae expression.
4. Tell users to add that common effect if they want a mean-zero group deviation that can be non-centered.
5. Never rewrite explicit exact or wildcard user priors.

This is the conservative option: it preserves current centered semantics, makes the non-Normal HDDM cases build, and prevents Normal orphan means. It intentionally changes the previous default-NC likelihood for unmatched Normal terms by reconnecting the generated location mean; that change should be explicit in the changelog.

The group-only branch should also use the semantic identity-link classifier from #1232, so omitted and explicit identity spellings cannot select different prior families.

### Proposed commit plan

This work should start after #1227 merges.

#### 1. `refactor: share effective noncentering with safe-prior generation (#1225)`

- Normalize/store the model-level boolean/dictionary setting once.
- Reuse a single precedence resolver: per-prior override, component-dictionary setting, then model default.
- Pass the effective configuration into `RegressionParam` safe-prior generation as well as diagnostics.
- Make no behavior change in this commit.
- Test booleans, component dictionaries, missing-key defaults, and per-prior precedence.

#### 2. `fix: center generated location-bearing group-only priors (#1225)`

- Identify structurally unmatched generated group terms through #1227's Formulae metadata.
- Preserve their current distribution family, bounds policy, and hyperparameters.
- Attach `noncentered=False` to the generated prior.
- When the effective model setting was non-centered, emit a term-specific warning that names:
  - the parameter;
  - the full group term;
  - the exact missing common expression;
  - the centered fallback HSSM applied.
- Preserve exact and `group_specific` wildcard priors without rewriting them.
- Leave matched #1224 terms zero-centered and parameterized as requested.
- Use semantic, not presence-based, identity-link classification.

#### 3. `fix: preflight unbuildable explicit group priors (#1225)`

- Inspect exact and wildcard user group priors structurally before Bambi builds.
- Under effective non-centering, raise one aggregated HSSM error for guaranteed build failures:
  - a hierarchical non-Normal outer prior;
  - a Normal prior with a free mean but no hierarchical `sigma`.
- Preserve #1227's warnings for Normal hierarchical-sigma cases that Bambi builds while ignoring/orphaning `mu`.
- Honor an explicit `noncentered=False` at per-prior or component level.
- Never rewrite user priors.

#### 4. `docs: define group-only prior semantics (#1225)`

- Explain the generated centered fallback and why zero is not a universal group-only location.
- Show how adding a common intercept/slope converts the group term to a zero-mean deviation eligible for non-centering.
- Document explicit-prior preflight errors and the precedence rules.
- Cross-link #1224/#1227 and Bambi #1003.
- Add an Unreleased changelog entry describing the intentional unmatched-Normal likelihood change.

### Acceptance matrix

- Generated unmatched group intercepts for `v`, `a`, `z`, `t`, `sv`, `sz`, `st`, and `p_outlier` build with model-level `noncentered=True` and `False`.
- Under model-level `True`, they use direct centered group RVs, emit the targeted fallback warning, and contain no orphan means.
- Prior families/hyperparameters match today's centered behavior.
- Matched safe terms retain #1224 behavior.
- Exact and wildcard explicit priors are never rewritten.
- Exact terms, wildcards, multiple grouping factors, and all parameterization-precedence levels are covered.
- Structure-only graph tests suffice; no sampling is required.

### Related boundaries

- #720 concerns no-intercept **common categorical cell means**, not group-only terms.
- #685 is related prior-display/parameterization context.
- Bambi #1003 tracks the upstream non-centered `mu` contract.
